acceptance/tests/resource/service/smf_basic_tests.rb

```diff
@@ -11,30 +11,35 @@
   end
 end
 
-
 agents.each do |agent|
   clean agent, :service => 'tstapp'
   manifest, method = setup agent, :service => 'tstapp'
-  step "SMF: clean slate"
-  apply_manifest_on(agent, 'service {tstapp : ensure=>stopped}') do
-    assert_match( /.*/, result.stdout, "err: #{agent}")
-  end
 
   step "SMF: ensure it is created with a manifest"
   apply_manifest_on(agent, 'service {tstapp : ensure=>running, manifest=>"%s"}' % manifest) do
-    assert_match( / ensure changed 'stopped'.* to 'running'/, result.stdout, "err: #{agent}")
+    assert_match( /defined 'ensure' as 'running'/, result.stdout, "err: #{agent}")
   end
-  step "SMF: verify it with puppet"
-  on(agent, puppet("resource service tstapp")) do
-    assert_match( /ensure => 'running'/, result.stdout, "err: #{agent}")
-  end
-
+  
   step "SMF: verify with svcs that the service is online"
   on agent, "svcs -l application/tstapp" do
     assert_match( /state\s+online/, result.stdout, "err: #{agent}")
   end
-  step "SMF: stop the service"
+  
+  step "SMF: ensure it is idempotent - ie not created again"
+  apply_manifest_on(agent, 'service {tstapp : ensure=>running, manifest=>"%s"}' % manifest, :catch_changes => true)
```

Comment:
yeah - actually I did that on purpose, with switching to `catch_changes => true`, which I think accomplishes the same thing without the fragility of the grep. IMO ideally I think we'd update all of them to work off change detection/error codes but the best way to go about that with the others was less clear cut to me and I want to get this thing green again..